server/app/Icepeak/Main.hs

```diff
@@ -77,33 +74,54 @@ runCore core = do
   httpServer <- HttpServer.new core
   let wsServer = WebsocketServer.acceptConnection core
 
+  let tryWithTag wrapper act = catch (act >> pure RCR_Fine) $ \exc -> case fromException exc of
```

Comment:
Please add a comment to explain why we do this. Maybe lift the function to the top-level for that, then you can add a type signature as well.

Reply:
Moved it to top level as:
```
-- | If a threads fails, catch the error and tag it so we know how to log it
catchRunCoreResult :: (SomeException -> RunCoreResult) -> IO () -> IO RunCoreResult
```